src/rotest/management/common/json_parser.py

```diff
@@ -0,0 +1,378 @@
+    parser's scheme. Failure in the validation case will cause a raise of
+    ParsingError exception.
+
+
+    Attributes:
+        complex_decoders (dict): map element type to its decoding method.
+    """
+    _NONE_TYPE = 'None'
+    _LIST_TYPE = 'List'
+    _CLASS_TYPE = 'Class'
+    _LIST_ITEM_TYPE = 'Item'
+    _DICT_TYPE = 'Dictionary'
+    _RESOURCE_TYPE = 'Resource'
+    _RESOURCE_DATA_TYPE = 'ResourceData'
+
+    def __init__(self):
+        super(JSONParser, self).__init__()
+        self.complex_decoders = {
+            self._LIST_TYPE: self._decode_list,
+            self._DICT_TYPE: self._decode_dict,
+            self._CLASS_TYPE: self._decode_class,
+            self._RESOURCE_TYPE: self._decode_resource,
+            self._RESOURCE_DATA_TYPE: self._decode_resource_data
+        }
+
+    def encode(self, resource_data):
+        """Encode a resource.
+
+        Args:
+            resource_data (ResourceData): a resource to encode.
+
+        Returns:
+            object. encoded data, depends on the parser type.
+
+        Raises:
+            TypeError: given 'resource' is not of type 'ResourceData'.
+            ParsingError: encoding failure.
+        """
+
+        return self._encode(resource_data)
+
+    def decode(self, data):
+        """Decode a message.
+
+        Args:
+            data (object): data to decode, encoded data of 'AbstractMessage',
+                depends on the parser type.
+
+        Returns:
+            AbstractMessage. decoded message.
+
+        Raises:
+            ParsingError: decoding failure.
+        """
+        # try:
+        return self._decode(data)
+        #
+        # except Exception as err:
+        #     raise ParsingError("Decoding data %r has failed. Reason: %s." %
+        #                        (data, err))
+
+    def _encode(self, data):
+        """Encode the given data according to its type.
+
+        Warning:
+            Do not change the order of the type validation! Some types matches
+            more than one type(e.g: bool is also a Number).
+
+        Args:
+            data (object): an object to encode.
+
+        Returns:
+            object. encoded data, depends on the parser type.
+
+        Raises:
+            TypeError: given 'data' couldn't be encoded by this parser.
+        """
+        if data is None:
+            return self._NONE_TYPE
+
+        if isinstance(data, basestring):
+            # Strings are surrounded by "" in order to prevent decoding errors.
+            # Without "", the string '5' will be wrongly decoded as number 5.
+            return data
+
+        if isinstance(data, bool):
+            # objectify identify only lower case (true / false) as bool type
+            return data
+
+        if isinstance(data, Number):
+            return data
+
+        if isinstance(data, dict):
+            return self._encode_dict(data)
+
+        if isinstance(data, (list, tuple)):
+            return self._encode_list(data)
```

Comment:
We really can't just use ```json``` for this whole functionality?

Reply:
i wanted it to be as  close as  possible to the original XML parser so i wouldn't have to do much changes.
I think it should be another  issue if  u  want  to refactored that.
It creates a complex structure and I  am  not sure  if it will break something.